connectors/src/lib/models.ts

```diff
@@ -149,6 +156,58 @@ SlackConfiguration.init(
 );
 Connector.hasOne(SlackConfiguration);
 
+export class NotionConnectorState extends Model<
+  InferAttributes<NotionConnectorState>,
+  InferCreationAttributes<NotionConnectorState>
+> {
+  declare id: CreationOptional<number>;
+  declare createdAt: CreationOptional<Date>;
+  declare updatedAt: CreationOptional<Date>;
+
+  declare lastGarbageCollectionStartTime?: Date;
+  declare lastGarbageCollectionFinishTime?: Date;
+
+  declare connectorId: ForeignKey<Connector["id"]>;
+}
+
+NotionConnectorState.init(
+  {
+    id: {
+      type: DataTypes.INTEGER,
+      autoIncrement: true,
+      primaryKey: true,
+    },
+    createdAt: {
+      type: DataTypes.DATE,
+      allowNull: false,
+      defaultValue: DataTypes.NOW,
+    },
+    updatedAt: {
+      type: DataTypes.DATE,
+      allowNull: false,
+      defaultValue: DataTypes.NOW,
+    },
+    lastGarbageCollectionStartTime: {
+      type: DataTypes.DATE,
+      allowNull: true,
+    },
+    lastGarbageCollectionFinishTime: {
+      type: DataTypes.DATE,
+      allowNull: true,
+    },
+  },
+  {
+    sequelize: sequelize_conn,
+    modelName: "notion_connector_states",
+    indexes: [{ fields: ["connectorId"], unique: true }],
+  }
+);
+
+Connector.hasOne(NotionConnectorState, {
+  as: "notionConnectorState",
+  foreignKey: "connectorId",
```

Comment:
not needed right?

Reply:
I don't know why but it seemed to be needed

Reply:
Well not needed for the SlackConfiguration. Maybe this is because you define option object?

Reply:
SlackConfiguration doesn't use the automatic joins feature of the ORM

Reply:
My point exactly ;-)